db-engine/src/main/java/com/splicemachine/db/iapi/services/property/PropertyUtil.java

```diff
@@ -92,7 +92,8 @@
 		"derby.language.logQueryPlan",
 		Property.LOCKS_ESCALATION_THRESHOLD,
 		Property.DATABASE_PROPERTIES_ONLY,
-		Property.DEFAULT_CONNECTION_MODE_PROPERTY
+		Property.DEFAULT_CONNECTION_MODE_PROPERTY,
+		Property.AUTHENTICATION_BUILTIN_ALGORITHM
```

Comment:
I don't quite understanding the relationship between adding this property and the main fix to allow first connection to go through without Kerberos authentication. Could you elaborate a bit?

Reply:
The first internal connection follows a different path from normal external connections. Usually the authentication algorithm is set on that other path, so for the first internal connection it wasn't set. By setting this here I make sure there's a suitable default for that first connection.

Reply:
Thanks Daniel for the explanation.